Updated to .NET core and latest Serilog.

xivk opened the pull request:
- Updated Serilog.
- Removed signing because datadog dependency isn't signed.
- Updated to new csproj format.

This should fix issues #8 #7 

I didn't fix the buildscript because I'm awful at powershell. I'm hoping someone is willing to update that part?

xivk committed:
Updated to .NET core and latest Serilog.
- Updated Serilog.
- Removed signing because datadog dependency isn't signed.
- Updated to new csproj format.

xivk committed:
Fixed assemblyname.

merbla commented:
@xivk try cherry picking [this](https://github.com/merbla/serilog-sinks-datadog/commit/9de5cadad155e5ed74f39c6e10c611fd1943fe30) commit.  

This is a quasi-standard build setup for sinks with .Net Core.

You should be able to iterate on that change to get the builds working.

xivk committed:
Updated buildscript basic on @merbla's suggestion.

xivk commented:
That was quick, thanks! :+1:  

Seems to be fixed (apart from my bad spelling in the commit message).

Perhaps the README should also be updated, it says .NET 4.5 now.

aregaz commented:
When can we expect updated nuget package with dotnet core 2.0 support?

nblumhardt commented on src/Serilog.Sinks.Datadog/Serilog.Sinks.Datadog.csproj:
```diff
@@ -1,78 +1,16 @@
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
+    <PackageId>Serilog.Sinks.Datadog</PackageId>
+    <Version>2.0.0</Version>
```
This needs to be a `<VersionPrefix>` and since we have 2.0.8 out there, might be good to rev the version number to 3.0.0.

nblumhardt reviewed (commented):
LGTM! Needs a slight tweak to how the package is versioned (comment above); might also be nice to list the new target frameworks instead of just .NET 4.5 in the README.md. Cheers!

xivk committed:
Added VersionPrefix to bump package version to 3.0.

xivk committed:
Updated platforms in README.

xivk commented:
Ok, changed the project file to include VersionPrefix and updated the README.

That did remind me of the fact that [the datadog dependency](https://www.nuget.org/packages/DogStatsD-CSharp-Client/) doesn't support 4.5 but it does support 4.5.1 so that's the minimum .NET framework version supported now.

xivk commented:
Also, shouldn't we sync the package version with the assembly version?

nblumhardt commented:
@xivk good point - since the package is no longer signed, we should probably just remove the assembly version and other information from the source and let the build system take care of it.

xivk commented:
Or just use `<Version>`? Or is that not an option? It's unclear for me how the publishing to nuget works so it's up to you.

The CustomAssemblyVersion file is already removed.

nblumhardt commented:
Thanks for the follow-up. `<Version>` won't work with the pre-release packaging scheme implemented in the build scripts. Hadn't noticed the custom assembly versioning removal - my bad, thanks!

xivk commented:
OK, thanks  :+1: 

Anything left for me to do on this?

nblumhardt commented:
Don't think so - thanks! :+1:
